micromamba/tests/test_create.py

```diff
@@ -652,6 +652,24 @@ def test_spec_with_channel_and_subdir():
         )
 
 
+def test_spec_with_slash_in_channel(tmp_home, tmp_root_prefix):
+    env_name = "myenv"
+    try:
+        res = helpers.create("-n", env_name, "pkgs/main/noarch::python", "--dry-run")
+    except subprocess.CalledProcessError as e:
+        assert e.stderr.decode() == (
+            'critical libmamba The package "pkgs/main/noarch::python" is '
+            "not available for the specified platform\n"
+        )
+
+    try:
+        os.environ["CONDA_SUBDIR"] = "linux-64"
+        helpers.create("-n", env_name, "pkgs/main/linux-64::python", "--dry-run")
+        helpers.create("-n", env_name, "pkgs/main::python", "--dry-run")
```

Comment:
I'm confused about this test. Is the first line supposed to succeed and not the second? Should both fail?

Reply:
Both should succeed with identical results, imo.

Reply:
In the regression we observed that `pkgs/main/linux-64::xxx` was accepted because the last component of the "channel path" was a known subdir, but `pkgs/main` failed because `main` is not a known subdir.

Reply:
I think the `try: finally:` would prevent the test to fail when these commands fail, no?

Anyhow, you can omit the `os.environ` reset. This is handled by pytest fixtures.